Auto generate index.theme

3v1n0 opened the pull request:
Do the same we're doing with accented themes.
I didn't include it in such branch as I wasn't sure if I was missing something :)

These are the differences with the static one we had

```diff
--- icons/Yaru/index.theme	2022-03-19 06:39:08.305942935 +0100
+++ /data/GNOME/_BUILD/yaru/icons/default-index.theme	2022-03-19 06:41:54.762321805 +0100
@@ -1,34 +1,38 @@
 [Icon Theme]
-Name=@ThemeName@
+Name=Yaru
 Comment=A desktop adaptation of the Ubuntu mobile icons.
 Inherits=Humanity,hicolor
 Example=folder
+Directories=24x24@2x/emblems,32x32@2x/actions,22x22@2x/legacy,48x48/devices,32x32/places,256x256/actions,16x16/status,32x32@2x/mimetypes,48x48/actions,24x24/legacy,48x48/emblems,16x16@2x/apps,256x256/legacy,24x24/devices,8x8@2x/status,256x256/status,8x8/actions,22x22@2x/categories,scalable/status,16x16/places,256x256/mimetypes,256x256/devices,22x22@2x/actions,scalable/ui,256x256@2x/mimetypes,48x48/apps,24x24@2x/places,16x16@2x/status,scalable/categories,32x32/status,16x16/legacy,48x48@2x/status,32x32/emblems,256x256/places,scalable/legacy,48x48@2x/actions,48x48/legacy,32x32@2x/devices,16x16@2x/places,scalable/mimetypes,32x32/mimetypes,32x32/categories,48x48/status,256x256@2x/places,22x22/legacy,32x32@2x/emblems,256x256@2x/emblems,24x24/places,24x24/mimetypes,32x32/legacy,16x16@2x/legacy,32x32@2x/categories,32x32/apps,22x22/categories,24x24/emblems,48x48@2x/emblems,48x48/categories,256x256@2x/devices,8x8/categories,8x8@2x/actions,16x16/mimetypes,48x48@2x/places,48x48/places,16x16/apps,256x256@2x/apps,256x256@2x/actions,scalable-max-32/status,24x24@2x/legacy,48x48@2x/apps,22x22/actions,scalable/apps,scalable/time,8x8/emblems,24x24/status,16x16@2x/devices,48x48@2x/legacy,256x256@2x/status,32x32/actions,8x8/status,24x24/apps,scalable/emotes,scalable/multimedia,8x8@2x/categories,16x16/devices,24x24@2x/apps,16x16@2x/categories,24x24@2x/status,24x24@2x/actions,16x16@2x/emblems,16x16/categories,32x32@2x/apps,16x16@2x/actions,48x48@2x/mimetypes,scalable/places,48x48@2x/devices,scalable/phosh,16x16@2x/mimetypes,24x24/actions,256x256@2x/categories,256x256@2x/legacy,32x32/devices,scalable/generic-symbols,scalable/actions,256x256/apps,24x24@2x/categories,32x32@2x/status,256x256/categories,scalable/camera,16x16/actions,scalable/emblems,24x24@2x/mimetypes,16x16/emblems,scalable/devices,32x32@2x/places,24x24@2x/devices,32x32@2x/legacy,256x256/emblems,48x48/mimetypes,8x8@2x/emblems,24x24/categories,48x48@2x/categories
 
-# Directory list
-Directories=8x8/emblems,16x16/actions,16x16/apps,16x16/categories,16x16/devices,16x16/emblems,16x16/emotes,16x16/legacy,16x16/mimetypes,16x16/panel,16x16/places,16x16/status,22x22/animations,22x22/panel,22x22/legacy,24x24/actions,24x24/animations,24x24/apps,24x24/categories,24x24/devices,24x24/emblems,24x24/emotes,24x24/legacy,24x24/mimetypes,24x24/panel,24x24/places,24x24/status,32x32/actions,32x32/apps,32x32/categories,32x32/devices,32x32/emblems,32x32/emotes,32x32/legacy,32x32/mimetypes,32x32/places,32x32/status,48x48/actions,48x48/apps,48x48/categories,48x48/devices,48x48/emblems,48x48/emotes,48x48/legacy,48x48/mimetypes,48x48/notifications,48x48/places,48x48/status,256x256/actions,256x256/apps,256x256/categories,256x256/devices,256x256/emblems,256x256/emotes,256x256/legacy,256x256/mimetypes,256x256/places,256x256/status,8x8@2x/emblems,16x16@2x/actions,16x16@2x/apps,16x16@2x/categories,16x16@2x/devices,16x16@2x/emblems,16x16@2x/emotes,16x16@2x/mimetypes,16x16@2x/panel,16x16@2x/places,16x16@2x/status,22x22@2x/animations,22x22@2x/panel,24x24@2x/actions,24x24@2x/animations,24x24@2x/apps,24x24@2x/categories,24x24@2x/devices,24x24@2x/emblems,24x24@2x/emotes,24x24@2x/mimetypes,24x24@2x/panel,24x24@2x/places,24x24@2x/status,32x32@2x/actions,32x32@2x/apps,32x32@2x/categories,32x32@2x/devices,32x32@2x/emblems,32x32@2x/emotes,32x32@2x/mimetypes,32x32@2x/places,32x32@2x/status,48x48@2x/actions,48x48@2x/apps,48x48@2x/categories,48x48@2x/devices,48x48@2x/emblems,48x48@2x/emotes,48x48@2x/mimetypes,48x48@2x/places,48x48@2x/status,256x256@2x/actions,256x256@2x/apps,256x256@2x/categories,256x256@2x/devices,256x256@2x/emblems,256x256@2x/emotes,256x256@2x/mimetypes,256x256@2x/places,256x256@2x/status,scalable/actions,scalable/apps,scalable/camera,scalable/categories,scalable/devices,scalable/emblems,scalable/emotes,scalable/generic-symbols,scalable/legacy,scalable/mimetypes,scalable/multimedia,scalable/phosh,scalable/places,scalable/status,scalable/time,scalable/ui,scalable-max-32/status
-
-# Actions
-
-[16x16/actions]
-Context=Actions
-Size=16
-Type=Fixed
-
-[24x24/actions]
-Context=Actions
+[24x24@2x/emblems]
+Context=Emblems
+Scale=2
 Size=24
 Type=Fixed
 
-[32x32/actions]
+[32x32@2x/actions]
 Context=Actions
+Scale=2
 Size=32
 Type=Fixed
 
-[48x48/actions]
-Context=Actions
+[22x22@2x/legacy]
+Context=Legacy
+Scale=2
+Size=22
+Type=Fixed
+
+[48x48/devices]
+Context=Devices
 Size=48
 Type=Fixed
 
+[32x32/places]
+Context=Places
+Size=32
+Type=Fixed
+
 [256x256/actions]
 Context=Actions
 Size=256
@@ -36,127 +40,85 @@
 MaxSize=256
 Type=Scalable
 
-[scalable/actions]
-Context=Actions
+[16x16/status]
+Context=Status
 Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-# Animations
-
-[22x22/animations]
-Context=Animations
-Size=22
 Type=Fixed
 
-[24x24/animations]
-Context=Animations
-Size=24
+[32x32@2x/mimetypes]
+Context=Mimetypes
+Scale=2
+Size=32
 Type=Fixed
 
-# Applications
-
-[16x16/apps]
-Context=Applications
-Size=16
+[48x48/actions]
+Context=Actions
+Size=48
 Type=Fixed
 
-[24x24/apps]
-Context=Applications
+[24x24/legacy]
+Context=Legacy
 Size=24
 Type=Fixed
 
-[32x32/apps]
-Context=Applications
-Size=32
+[48x48/emblems]
+Context=Emblems
+Size=48
 Type=Fixed
 
-[48x48/apps]
+[16x16@2x/apps]
 Context=Applications
-Size=48
+Scale=2
+Size=16
 Type=Fixed
 
-[256x256/apps]
-Context=Applications
+[256x256/legacy]
+Context=Legacy
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-[scalable/apps]
-Context=Applications
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-# Camera
-
-[scalable/camera]
-Context=Camera
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-# Categories
-
-[16x16/categories]
-Context=Categories
-Size=16
-Type=Fixed
-
-[24x24/categories]
-Context=Categories
+[24x24/devices]
+Context=Devices
 Size=24
 Type=Fixed
 
-[32x32/categories]
-Context=Categories
-Size=32
-Type=Fixed
-
-[48x48/categories]
-Context=Categories
-Size=48
+[8x8@2x/status]
+Context=Status
+Scale=2
+Size=8
 Type=Fixed
 
-[256x256/categories]
-Context=Categories
+[256x256/status]
+Context=Status
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-[scalable/categories]
-Context=Categories
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-# Devices
-
-[16x16/devices]
-Context=Devices
-Size=16
+[8x8/actions]
+Context=Actions
+Size=8
 Type=Fixed
 
-[24x24/devices]
-Context=Devices
-Size=24
+[22x22@2x/categories]
+Context=Categories
+Scale=2
+Size=22
 Type=Fixed
 
-[32x32/devices]
-Context=Devices
-Size=32
+[16x16/places]
+Context=Places
+Size=16
 Type=Fixed
 
-[48x48/devices]
-Context=Devices
-Size=48
-Type=Fixed
+[256x256/mimetypes]
+Context=Mimetypes
+Size=256
+MinSize=64
+MaxSize=256
+Type=Scalable
 
 [256x256/devices]
 Context=Devices
@@ -165,665 +127,489 @@
 MaxSize=256
 Type=Scalable
 
-[scalable/devices]
-Context=Devices
-Size=16
-MinSize=16
+[22x22@2x/actions]
+Context=Actions
+Scale=2
+Size=22
+Type=Fixed
+
+[256x256@2x/mimetypes]
+Context=Mimetypes
+Scale=2
+Size=256
+MinSize=64
 MaxSize=256
 Type=Scalable
 
-# Emblems
+[48x48/apps]
+Context=Applications
+Size=48
+Type=Fixed
 
-[8x8/emblems]
-Context=Emblems
-Size=8
+[24x24@2x/places]
+Context=Places
+Scale=2
+Size=24
 Type=Fixed
 
-[16x16/emblems]
-Context=Emblems
+[16x16@2x/status]
+Context=Status
+Scale=2
 Size=16
 Type=Fixed
 
-[24x24/emblems]
-Context=Emblems
-Size=24
+[32x32/status]
+Context=Status
+Size=32
 Type=Fixed
 
-[32x32/emblems]
-Context=Emblems
-Size=32
+[16x16/legacy]
+Context=Legacy
+Size=16
 Type=Fixed
 
-[48x48/emblems]
-Context=Emblems
+[48x48@2x/status]
+Context=Status
+Scale=2
 Size=48
 Type=Fixed
 
-[256x256/emblems]
+[32x32/emblems]
 Context=Emblems
+Size=32
+Type=Fixed
+
+[256x256/places]
+Context=Places
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-[scalable/emblems]
-Context=Emblems
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
+[48x48@2x/actions]
+Context=Actions
+Scale=2
+Size=48
+Type=Fixed
 
-# Emotes
+[48x48/legacy]
+Context=Legacy
+Size=48
+Type=Fixed
 
-[16x16/emotes]
-Context=Emotes
+[32x32@2x/devices]
+Context=Devices
+Scale=2
+Size=32
+Type=Fixed
+
+[16x16@2x/places]
+Context=Places
+Scale=2
 Size=16
 Type=Fixed
 
-[24x24/emotes]
-Context=Emotes
-Size=24
+[32x32/mimetypes]
+Context=Mimetypes
+Size=32
 Type=Fixed
 
-[32x32/emotes]
-Context=Emotes
+[32x32/categories]
+Context=Categories
 Size=32
 Type=Fixed
 
-[48x48/emotes]
-Context=Emotes
+[48x48/status]
+Context=Status
 Size=48
 Type=Fixed
 
-[256x256/emotes]
-Context=Emotes
+[256x256@2x/places]
+Context=Places
+Scale=2
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-[scalable/emotes]
-Context=Emotes
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-# Legacy
-
-[16x16/legacy]
-Context=Legacy
-Size=16
-Type=Fixed
-
 [22x22/legacy]
 Context=Legacy
 Size=22
 Type=Fixed
 
-[24x24/legacy]
-Context=Legacy
-Size=24
-Type=Fixed
-
-[32x32/legacy]
-Context=Legacy
+[32x32@2x/emblems]
+Context=Emblems
+Scale=2
 Size=32
 Type=Fixed
 
-[48x48/legacy]
-Context=Legacy
-Size=48
-Type=Fixed
-
-[256x256/legacy]
-Context=Legacy
+[256x256@2x/emblems]
+Context=Emblems
+Scale=2
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-[scalable/legacy]
-Context=Legacy
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-# Generic symbols
-
-[scalable/generic-symbols]
-Context=generic-symbols
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-# Multimedia
-
-[scalable/multimedia]
-Context=Multimedia
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-# MimeTypes
-
-[16x16/mimetypes]
-Context=MimeTypes
-Size=16
+[24x24/places]
+Context=Places
+Size=24
 Type=Fixed
 
 [24x24/mimetypes]
-Context=MimeTypes
+Context=Mimetypes
 Size=24
 Type=Fixed
 
-[32x32/mimetypes]
-Context=MimeTypes
+[32x32/legacy]
+Context=Legacy
 Size=32
 Type=Fixed
 
-[48x48/mimetypes]
-Context=MimeTypes
-Size=48
-Type=Fixed
-
-[256x256/mimetypes]
-Context=MimeTypes
-Size=256
-MinSize=64
-MaxSize=256
-Type=Scalable
-
-[scalable/mimetypes]
-Context=Mimetypes
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-# Notifications
-
-[48x48/notifications]
-Context=Status
-Size=48
-Type=Fixed
-
-
-# Panel
-
-[16x16/panel]
-Context=Status
+[16x16@2x/legacy]
+Context=Legacy
+Scale=2
 Size=16
 Type=Fixed
 
-[22x22/panel]
-Context=Status
-Size=22
+[32x32@2x/categories]
+Context=Categories
+Scale=2
+Size=32
 Type=Fixed
 
-[24x24/panel]
-Context=Status
-Size=24
+[32x32/apps]
+Context=Applications
+Size=32
 Type=Fixed
 
-# Phosh
-[scalable/phosh]
-Context=Phosh
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-# Places
-
-[16x16/places]
-Context=Places
-Size=16
+[22x22/categories]
+Context=Categories
+Size=22
 Type=Fixed
 
-[24x24/places]
-Context=Places
+[24x24/emblems]
+Context=Emblems
 Size=24
 Type=Fixed
 
-[32x32/places]
-Context=Places
-Size=32
+[48x48@2x/emblems]
+Context=Emblems
+Scale=2
+Size=48
 Type=Fixed
 
-[48x48/places]
-Context=Places
+[48x48/categories]
+Context=Categories
 Size=48
 Type=Fixed
 
-[256x256/places]
-Context=Places
+[256x256@2x/devices]
+Context=Devices
+Scale=2
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-[scalable/places]
-Context=Places
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
+[8x8/categories]
+Context=Categories
+Size=8
+Type=Fixed
 
-# Status
+[8x8@2x/actions]
+Context=Actions
+Scale=2
+Size=8
+Type=Fixed
 
-[16x16/status]
-Context=Status
+[16x16/mimetypes]
+Context=Mimetypes
 Size=16
 Type=Fixed
 
-[24x24/status]
-Context=Status
-Size=24
+[48x48@2x/places]
+Context=Places
+Scale=2
+Size=48
 Type=Fixed
 
-[32x32/status]
-Context=Status
-Size=32
+[48x48/places]
+Context=Places
+Size=48
 Type=Fixed
 
-[48x48/status]
-Context=Status
-Size=48
+[16x16/apps]
+Context=Applications
+Size=16
 Type=Fixed
 
-[256x256/status]
-Context=Status
+[256x256@2x/apps]
+Context=Applications
+Scale=2
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-[scalable/status]
-Context=Status
-Size=16
-MinSize=16
-MaxSize=256
-Type=Scalable
-
-[scalable-max-32/status]
-Context=Status
-Size=16
-MinSize=16
-MaxSize=32
-Type=Scalable
-
-# time
-
-[scalable/time]
-Context=time
-Size=16
-MinSize=16
+[256x256@2x/actions]
+Context=Actions
+Scale=2
+Size=256
+MinSize=64
 MaxSize=256
 Type=Scalable
 
-# UI
-
-[scalable/ui]
-Context=UI
-Size=16
-MinSize=8
-MaxSize=512
-Type=Scalable
-
-
-#
-# HiDPi @2x scale
-#
-
-
-# Actions
+[24x24@2x/legacy]
+Context=Legacy
+Scale=2
+Size=24
+Type=Fixed
 
-[16x16@2x/actions]
-Context=Actions
+[48x48@2x/apps]
+Context=Applications
 Scale=2
-Size=16
+Size=48
 Type=Fixed
 
-[24x24@2x/actions]
+[22x22/actions]
 Context=Actions
-Scale=2
+Size=22
+Type=Fixed
+
+[8x8/emblems]
+Context=Emblems
+Size=8
+Type=Fixed
+
+[24x24/status]
+Context=Status
 Size=24
 Type=Fixed
 
-[32x32@2x/actions]
-Context=Actions
+[16x16@2x/devices]
+Context=Devices
 Scale=2
-Size=32
+Size=16
 Type=Fixed
 
-[48x48@2x/actions]
-Context=Actions
+[48x48@2x/legacy]
+Context=Legacy
 Scale=2
 Size=48
 Type=Fixed
 
-[256x256@2x/actions]
-Context=Actions
+[256x256@2x/status]
+Context=Status
 Scale=2
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-# Animations
-
-[22x22@2x/animations]
-Context=Animations
-Scale=2
-Size=22
+[32x32/actions]
+Context=Actions
+Size=32
 Type=Fixed
 
-[24x24@2x/animations]
-Context=Animations
-Scale=2
-Size=24
+[8x8/status]
+Context=Status
+Size=8
 Type=Fixed
 
-
-# Applications
-
-[16x16@2x/apps]
+[24x24/apps]
 Context=Applications
-Scale=2
-Size=16
+Size=24
 Type=Fixed
 
-[24x24@2x/apps]
-Context=Applications
+[8x8@2x/categories]
+Context=Categories
 Scale=2
-Size=24
+Size=8
 Type=Fixed
 
-[32x32@2x/apps]
-Context=Applications
-Scale=2
-Size=32
+[16x16/devices]
+Context=Devices
+Size=16
 Type=Fixed
 
-[48x48@2x/apps]
+[24x24@2x/apps]
 Context=Applications
 Scale=2
-Size=48
+Size=24
 Type=Fixed
 
-[256x256@2x/apps]
-Context=Apps
-Scale=2
-Size=256
-MinSize=64
-MaxSize=256
-Type=Scalable
-
-# Categories
-
 [16x16@2x/categories]
 Context=Categories
 Scale=2
 Size=16
 Type=Fixed
 
-[24x24@2x/categories]
-Context=Categories
+[24x24@2x/status]
+Context=Status
 Scale=2
 Size=24
 Type=Fixed
 
-[32x32@2x/categories]
-Context=Categories
+[24x24@2x/actions]
+Context=Actions
 Scale=2
-Size=32
+Size=24
 Type=Fixed
 
-[48x48@2x/categories]
-Context=Categories
+[16x16@2x/emblems]
+Context=Emblems
 Scale=2
-Size=48
+Size=16
 Type=Fixed
 
-[256x256@2x/categories]
+[16x16/categories]
 Context=Categories
-Scale=2
-Size=256
-MinSize=64
-MaxSize=256
-Type=Scalable
-
-# Devices
-
-[16x16@2x/devices]
-Context=Devices
-Scale=2
 Size=16
 Type=Fixed
 
-[24x24@2x/devices]
-Context=Devices
+[32x32@2x/apps]
+Context=Applications
 Scale=2
-Size=24
+Size=32
 Type=Fixed
 
-[32x32@2x/devices]
-Context=Devices
+[16x16@2x/actions]
+Context=Actions
 Scale=2
-Size=32
+Size=16
 Type=Fixed
 
-[48x48@2x/devices]
-Context=Devices
+[48x48@2x/mimetypes]
+Context=Mimetypes
 Scale=2
 Size=48
 Type=Fixed
 
-[256x256@2x/devices]
+[48x48@2x/devices]
 Context=Devices
 Scale=2
-Size=256
-MinSize=64
-MaxSize=256
-Type=Scalable
-
-
-# Emblems
-
-[8x8@2x/emblems]
-Context=Emblems
-Scale=2
-Size=8
+Size=48
 Type=Fixed
 
-[16x16@2x/emblems]
-Context=Emblems
+[16x16@2x/mimetypes]
+Context=Mimetypes
 Scale=2
 Size=16
 Type=Fixed
 
-[24x24@2x/emblems]
-Context=Emblems
-Scale=2
+[24x24/actions]
+Context=Actions
 Size=24
 Type=Fixed
 
-[32x32@2x/emblems]
-Context=Emblems
-Scale=2
-Size=32
-Type=Fixed
-
-[48x48@2x/emblems]
-Context=Emblems
-Scale=2
-Size=48
-Type=Fixed
-
-[256x256@2x/emblems]
-Context=Emblems
+[256x256@2x/categories]
+Context=Categories
 Scale=2
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-# Emotes
-
-[16x16@2x/emotes]
-Context=Emotes
-Scale=2
-Size=16
-Type=Fixed
-
-[24x24@2x/emotes]
-Context=Emotes
+[256x256@2x/legacy]
+Context=Legacy
 Scale=2
-Size=24
-Type=Fixed
+Size=256
+MinSize=64
+MaxSize=256
+Type=Scalable
 
-[32x32@2x/emotes]
-Context=Emotes
-Scale=2
+[32x32/devices]
+Context=Devices
 Size=32
 Type=Fixed
 
-[48x48@2x/emotes]
-Context=Emotes
-Scale=2
-Size=48
-Type=Fixed
-
-[256x256@2x/emotes]
-Context=Emotes
-Scale=2
+[256x256/apps]
+Context=Applications
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-# MimeTypes
-
-[16x16@2x/mimetypes]
-Context=MimeTypes
-Scale=2
-Size=16
-Type=Fixed
-
-[24x24@2x/mimetypes]
-Context=MimeTypes
+[24x24@2x/categories]
+Context=Categories
 Scale=2
 Size=24
 Type=Fixed
 
-[32x32@2x/mimetypes]
-Context=MimeTypes
+[32x32@2x/status]
+Context=Status
 Scale=2
 Size=32
 Type=Fixed
 
-[48x48@2x/mimetypes]
-Context=MimeTypes
-Scale=2
-Size=48
-Type=Fixed
-
-[256x256@2x/mimetypes]
-Context=Mimetypes
-Scale=2
+[256x256/categories]
+Context=Categories
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-# Panel
-
-[16x16@2x/panel]
-Context=Status
-Scale=2
+[16x16/actions]
+Context=Actions
 Size=16
 Type=Fixed
 
-[22x22@2x/panel]
-Context=Status
-Scale=2
-Size=22
-Type=Fixed
-
-[24x24@2x/panel]
-Context=Status
+[24x24@2x/mimetypes]
+Context=Mimetypes
 Scale=2
 Size=24
 Type=Fixed
 
-# Places
-
-[16x16@2x/places]
-Context=Places
-Scale=2
+[16x16/emblems]
+Context=Emblems
 Size=16
 Type=Fixed
 
-[24x24@2x/places]
-Context=Places
-Scale=2
-Size=24
-Type=Fixed
-
 [32x32@2x/places]
 Context=Places
 Scale=2
 Size=32
 Type=Fixed
 
-[48x48@2x/places]
-Context=Places
+[24x24@2x/devices]
+Context=Devices
 Scale=2
-Size=48
+Size=24
 Type=Fixed
 
-[256x256@2x/places]
-Context=Places
+[32x32@2x/legacy]
+Context=Legacy
 Scale=2
+Size=32
+Type=Fixed
+
+[256x256/emblems]
+Context=Emblems
 Size=256
 MinSize=64
 MaxSize=256
 Type=Scalable
 
-# Status
-
-[16x16@2x/status]
-Context=Status
-Scale=2
-Size=16
+[48x48/mimetypes]
+Context=Mimetypes
+Size=48
 Type=Fixed
 
-[24x24@2x/status]
-Context=Status
+[8x8@2x/emblems]
+Context=Emblems
 Scale=2
-Size=24
+Size=8
 Type=Fixed
 
-[32x32@2x/status]
-Context=Status
-Scale=2
-Size=32
+[24x24/categories]
+Context=Categories
+Size=24
 Type=Fixed
 
-[48x48@2x/status]
-Context=Status
+[48x48@2x/categories]
+Context=Categories
 Scale=2
 Size=48
 Type=Fixed
 
-[256x256@2x/status]
-Context=Status
-Scale=2
-Size=256
-MinSize=64
-MaxSize=256
-Type=Scalable
```

Jupi007 commented:
Those new automations are awesome @3v1n0 !

It's very hard to review such a large diff, but at the first look I can say it miss the sizes definitions of symbolic icons.
I didn't checked everything but the fullcolor ones seems all there.
Also I don't know if it's still needed, but I remember let the mate variant inherit the default one.

3v1n0 commented:
> Also I don't know if it's still needed, but I remember let the mate variant inherit the default one. Yeah this is happening as per https://github.com/ubuntu/yaru/blob/22.04.2/icons/meson.build#L168

> but at the first look I can say it miss the sizes definitions of symbolic icons.

Wasn't this enough? What else do you think is missing?

https://github.com/ubuntu/yaru/pull/3504/commits/4697d33f00e2aa7d47b6522075d606b5deded951

3v1n0 commented:
Ok, sorry I think that I had a logic error...

This now fixed.

I've been through comparing the two using this dirty script, and everything relevant matches now (there were some errors and wrong elements in the static file, so I needed to filter them out):

```python
import configparser
import sys

orig = configparser.ConfigParser()
new = configparser.ConfigParser()

# These were added to old theme, but are not valid anymore
IGNORED_CONTEXTS = [
    'animations',
    'panel',
    'generic-icons',
    'notifications',
]

IGNORED_SIZE_CONTEXTS = {
    '16x16': ['emotes'],
    '16x16@2x': ['emotes'],
    '24x24': ['emotes'],
    '24x24@2x': ['emotes'],
    '32x32': ['emotes'],
    '32x32@2x': ['emotes'],
    '48x48': ['emotes'],
    '48x48@2x': ['emotes'],
    '256x256': ['emotes'],
    '256x256@2x': ['emotes'],
}

orig.read(sys.argv[1])
new.read(sys.argv[2])

def check_sections(a, b, do_assert=True):
    for s in a.sections():
        if '/' in s:
            [size, context] = s.split('/')

            if context in IGNORED_CONTEXTS or context in IGNORED_SIZE_CONTEXTS.get(size, []):
                continue

        print(s)

        if do_assert:
            assert s in b.sections()
        elif s not in b.sections():
            print(f'  Section {s} is missing!')

def check_icons_sections_deep(a, b, do_assert=True):
    for s in a.sections():
        if not '/' in s or s not in b.sections():
            continue

        print(f'[{s}]')
        for key in a[s]:
            av = a[s][key]
            bv = b[s].get(key, None)
            print(f'{key} = {av}, VS has {bv}')
            assert av == bv

check_sections(orig, new)
check_sections(new, orig, do_assert=False)
check_icons_sections_deep(orig, new)
```
